pkg/common/util/signals.go

```diff
@@ -2,20 +2,17 @@ package util
 
 import (
 	"context"
-	"os"
 	"os/signal"
 	"syscall"
 )
 
-func SignalListener(ctx context.Context, cancel func()) {
+func SignalListener(parent context.Context, cancel func()) {
 	go func() {
-		signalCh := make(chan os.Signal, 1)
-		signal.Notify(signalCh, syscall.SIGINT, syscall.SIGTERM)
+		ctx, cancel := signal.NotifyContext(parent, syscall.SIGINT, syscall.SIGTERM)
 
 		select {
+		case <-parent.Done():
```

Comment:
This select case is unnecessary since if `parent` is cancelled, `ctx` will be cancelled too.

Reply:
I removed `utils.SignalListener`.